hello/logo_grab.py

```diff
@@ -0,0 +1,52 @@
+import requests
+from bs4 import BeautifulSoup
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urllib2 import urlparse
+    urlparse = urlparse.urlparse
+
+try:
+    ConnectionError
+except NameError:
+    ConnectionError = ValueError
+
+YAHOO_ENDPOINT = "http://finance.yahoo.com/q/pr?s={}"
+CLEARBIT_ENDPOINT = "https://logo.clearbit.com/{}?format=png&size=438"
+
+def get_endpoint(ticker):
+    return YAHOO_ENDPOINT.format(ticker).lower()
+
+
+def get_response(target):
+    return requests.get(target)
+
+
+def handle_response(response):
+    if not response.ok:
+        raise ConnectionError("Yahoo didn't like that")
+    soup = BeautifulSoup(response.text, "html5lib")
+    pool = []
+    for anchor in soup.find_all("a", href=True):
+        link = anchor['href'].lower()
+        if link.startswith("http://") or link.startswith("https://"):
+            if link in pool:
+                return link
+            if "yahoo" not in link:
+                pool.append(link)
+    raise ValueError("Invalid Ticker")
```

Comment:
I'd love a docstring that explains exactly what this function exists for.  It seems like it returns the first found http or https url that does not refer to yahoo itself so long as that same url is found twice.  Why?  this is the kind of function that makes perfect sense to you when you write it, but when you return three months later, you can't remember why it exists.  Docstrings (or comments) would help fill the memory gap on the purpose.